vm/src/stdlib/itertools.rs

```diff
@@ -682,13 +683,24 @@ mod decl {
         }
     }
 
-    fn pyobject_to_opt_usize(obj: PyObjectRef, vm: &VirtualMachine) -> Option<usize> {
+    // Restrict obj to ints with value 0 <= val <= sys.maxsize (isize::MAX).
+    // On failure (out of range, non-int object) a ValueError is raised.
+    fn pyobject_to_opt_usize(obj: PyObjectRef, name: &str, vm: &VirtualMachine) -> PyResult<usize> {
         let is_int = obj.isinstance(&vm.ctx.types.int_type);
         if is_int {
-            int::get_value(&obj).to_usize()
-        } else {
-            None
+            let value = int::get_value(&obj).to_usize();
+            if let Some(value) = value {
+                // Only succeeds for values for which 0 <= value <= isize::MAX
+                if value <= usize::try_from(isize::MAX).unwrap() {
```

Comment:
Because `isize::MAX` is a const, I think `isize::MAX as usize` or `usize::try_from(isize::MAX).unwrap_or_else(|| unsafe { std::hint::unreachable_unchecked() })` is safe enough

Reply:
Thanks for the feedback, I went with your first suggestion.